infra/modules/providers/azure/app-service/main.tf

```diff
@@ -14,25 +17,34 @@ data "azurerm_app_service_plan" "appsvc" {
 }
 
 resource "azurerm_app_service" "appsvc" {
-  name                = "${lower(element(keys(var.app_service_name), count.index))}-${lower(terraform.workspace)}"
+  name                = "${lower(element(keys(var.app_service_config), count.index))}-${lower(terraform.workspace)}"
   resource_group_name = data.azurerm_resource_group.appsvc.name
   location            = data.azurerm_resource_group.appsvc.location
   app_service_plan_id = data.azurerm_app_service_plan.appsvc.id
   tags                = var.resource_tags
-  count               = length(keys(var.app_service_name))
+  count               = length(keys(var.app_service_config))
 
   app_settings = {
-    DOCKER_REGISTRY_SERVER_URL          = "https://${var.docker_registry_server_url}"
+    DOCKER_REGISTRY_SERVER_URL          = format("https://%s", var.docker_registry_server_url)
     WEBSITES_ENABLE_APP_SERVICE_STORAGE = false
     DOCKER_REGISTRY_SERVER_USERNAME     = var.docker_registry_server_username
     DOCKER_REGISTRY_SERVER_PASSWORD     = var.docker_registry_server_password
     APPINSIGHTS_INSTRUMENTATIONKEY      = var.app_insights_instrumentation_key
     KEYVAULT_URI                        = var.vault_uri
     DOCKER_ENABLE_CI                    = var.docker_enable_ci
+  }        
+
+  auth_settings {
+    enabled            = var.enable_auth
```

Comment:
Could this be inferred off of the value of `ad_client_id` such that we can reduce the # of variables for the template to a minimum? It may look something like this:
`locals.app_configs[count.index].ad_client_id == "" ? false : true`

Reply:
Your suggestion makes way more sense because it scopes authorization to the app level and away from being a master switch that enables authentication for all. Thanks!!